From 1718d11005da4bd288aaf84ac3965dc490e16278 Mon Sep 17 00:00:00 2001
From: Alexander Khryukin <alexander@mezon.ru>
Date: Thu, 16 Dec 2010 09:57:58 -0800
Subject: [PATCH 10/20] OSAL: fix OSAL_wcsmp() for Linux environments

Good day, Inaky. This is patch for wimaxd that allow connect to wimax
network Yota from Carrick (part of connmand).

in InfraStack/OSAgnostic/WiMax/Wrappers/CommonAPI/wimaxapi.c in
function CmdConnectToNetwork() call to OSAL_wcscmp() with
nspName="Yota" and nspsEx[0].NSPName="Yota" returns -1246756864
---
 .../Linux/OSAL/Primitives/wimax_osal_strings.c     |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/InfraStack/OSDependent/Linux/OSAL/Primitives/wimax_osal_strings.c b/InfraStack/OSDependent/Linux/OSAL/Primitives/wimax_osal_strings.c
index f97119a..6a71b11 100644
--- a/InfraStack/OSDependent/Linux/OSAL/Primitives/wimax_osal_strings.c
+++ b/InfraStack/OSDependent/Linux/OSAL/Primitives/wimax_osal_strings.c
@@ -1653,9 +1653,13 @@ int OSAL_CAPI_wmemcpy_s(OSAL_WIMAX_API_WSTRING dest, size_t sizeInWords, const O
 }
 // TODO::OSAL-WORK::IMPORTANT - OSAL_wcscmp, OSAL_wcslen - What happens when these methods are fed with ansi strings???
 
-int OSAL_wcscmp(const wchar_t * string1, const wchar_t * string2)
+int OSAL_wcscmp(OSAL_WIMAX_API_WSTRING string1, OSAL_WIMAX_API_WSTRING string2)
 {
+#ifdef NATIVE_LINUX
+	return strcmp(string1, string2);
+#else
 	return wcscmp(string1, string2);
+#endif
 }
 
 
-- 
1.7.4.1

